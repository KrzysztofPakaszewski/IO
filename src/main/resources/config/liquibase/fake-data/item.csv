id;title;state;category;image;image_content_type;hash;preferences;prefered_delivery
1;South Dakota envisioneer Music;Nakfa;Books;../fake-data/blob/hipster.png;image/png;Salad;Digitized invoice;Courier
2;Georgia Industrial;connect Markets Chief;Games;../fake-data/blob/hipster.png;image/png;Aruba;software scale interactive;Courier
3;scalable;Small;Movies;../fake-data/blob/hipster.png;image/png;magenta monitor 1080p;Unbranded Plastic Tuna Frozen;InPost
4;Lempira;pixel withdrawal;Books;../fake-data/blob/hipster.png;image/png;hard drive Robust Networked;Poland;Courier
5;Dynamic;knowledge base Buckinghamshire;Movies;../fake-data/blob/hipster.png;image/png;Brand;neutral Salad;InPost
6;mint green e-business EXE;back-end;Games;../fake-data/blob/hipster.png;image/png;Jewelery cutting-edge;Games;PersonalCollection
7;encompassing;Directives;Games;../fake-data/blob/hipster.png;image/png;facilitate Tasty Concrete Chips Exclusive;Buckinghamshire capability zero tolerance;PersonalCollection
8;paradigms Home;Investor Public-key;Games;../fake-data/blob/hipster.png;image/png;Dominican Republic Credit Card Account Handmade;Ergonomic transmitter;InPost
9;bypassing optical;installation;Books;../fake-data/blob/hipster.png;image/png;neural-net red;Buckinghamshire Future-proofed deposit;InPost
10;Cambridgeshire hacking;bricks-and-clicks Concrete Global;Books;../fake-data/blob/hipster.png;image/png;Borders content-based;Tasty Customer Human;PersonalCollection
