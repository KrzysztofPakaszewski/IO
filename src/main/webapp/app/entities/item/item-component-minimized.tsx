import React from 'react';
import Card from '@material-ui/core/Card';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import { Button} from 'reactstrap';
import { Link } from 'react-router-dom';


export const ItemCompMinimized = (itemEntity) =>{

  return (
    <Card>
      <Button color="link" tag={Link} to={`/search/${itemEntity.id}`}>
        <Grid container
          direction="column"
          justify="space-around"
          alignItems="center"
        >
          <Grid item>
              {itemEntity.image && (
                <img src={`data:${itemEntity.imageContentType};base64,${itemEntity.image}`} height = '80%' width=  '50%' />
              )}
          </Grid>
          <Grid item>
            {itemEntity.title}
          </Grid>
        </Grid>
      </Button>
    </Card>
  );
}
