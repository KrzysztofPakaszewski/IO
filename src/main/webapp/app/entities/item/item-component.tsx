import React from 'react';
import Rating from '@material-ui/lab/Rating';
import Card from '@material-ui/core/Card';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import { Button} from 'reactstrap';
import { Link } from 'react-router-dom';
import {openFile} from "react-jhipster";


// props: {score:Integer(0-5), login:String, review:String}
export const ItemComp = (itemEntity) =>{

  return (
    <Box maxWidth={600}>
        <dd>
          {itemEntity.image ? (
          <div>
            <a onClick={openFile(itemEntity.imageContentType, itemEntity.image)}>
              <img src={`data:${itemEntity.imageContentType};base64,${itemEntity.image}`} height = '300' width=  '200' />
            </a>
          </div>
        ) : null}
        </dd>
        <dt>
          <span id="title">Title</span>
        </dt>
        <dd>{itemEntity.title}</dd>
        <dt>
          <span id="category">Category</span>
        </dt>
        <dd>{itemEntity.category}</dd>
        <dt>
          <span id="state">State</span>
        </dt>
        <dd>{itemEntity.state}</dd>
        <dt>
          <span id="preferedDelivery">Prefered Delivery</span>
        </dt>
        <dd>{itemEntity.preferedDelivery}</dd>
        <dt>
          <span id="preferences">Preferences</span>
        </dt>
        <dd>{itemEntity.preferences}</dd>
        <dt>
          <span id="hash">Hashtags</span>
        </dt>
        <dd>{itemEntity.hash}</dd>
    </Box>
  );
}
