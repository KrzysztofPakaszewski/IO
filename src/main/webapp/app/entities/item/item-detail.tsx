import React from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps } from 'react-router-dom';
import { Button, Row, Col } from 'reactstrap';
import { ICrudGetAction, openFile, byteSize } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { IRootState } from 'app/shared/reducers';
import { getEntity } from './item.reducer';
import { IItem } from 'app/shared/model/item.model';
import { APP_DATE_FORMAT, APP_LOCAL_DATE_FORMAT } from 'app/config/constants';

export interface IItemDetailProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string }> {}

export class ItemDetail extends React.Component<IItemDetailProps> {
  componentDidMount() {
    this.props.getEntity(this.props.match.params.id);
  }

  render() {
    const { itemEntity } = this.props;
    return (
      <Row>
        <Col md="8">
          <h2>
            Item [<b>{itemEntity.id}</b>]
          </h2>
          <dl className="jh-entity-details">
            <dt>
              <span id="title">Title</span>
            </dt>
            <dd>{itemEntity.title}</dd>
            <dt>
              <span id="state">State</span>
            </dt>
            <dd>{itemEntity.state}</dd>
            <dt>
              <span id="category">Category</span>
            </dt>
            <dd>{itemEntity.category}</dd>
            <dt>
              <span id="image">Image</span>
            </dt>
            <dd>
              {itemEntity.image ? (
                <div>
                  <a onClick={openFile(itemEntity.imageContentType, itemEntity.image)}>
                    <img src={`data:${itemEntity.imageContentType};base64,${itemEntity.image}`} style={{ maxHeight: '30px' }} />
                  </a>
                  <span>
                    {itemEntity.imageContentType}, {byteSize(itemEntity.image)}
                  </span>
                </div>
              ) : null}
            </dd>
            <dt>
              <span id="hash">Hash</span>
            </dt>
            <dd>{itemEntity.hash}</dd>
            <dt>
              <span id="preferences">Preferences</span>
            </dt>
            <dd>{itemEntity.preferences}</dd>
            <dt>
              <span id="preferedDelivery">Prefered Delivery</span>
            </dt>
            <dd>{itemEntity.preferedDelivery}</dd>
            <dt>Owner</dt>
            <dd>{itemEntity.owner ? itemEntity.owner.id : ''}</dd>
          </dl>
          <Button tag={Link} to="/item" replace color="info">
            <FontAwesomeIcon icon="arrow-left" /> <span className="d-none d-md-inline">Back</span>
          </Button>
          &nbsp;
          <Button tag={Link} to={`/item/${itemEntity.id}/edit`} replace color="primary">
            <FontAwesomeIcon icon="pencil-alt" /> <span className="d-none d-md-inline">Edit</span>
          </Button>
        </Col>
      </Row>
    );
  }
}

const mapStateToProps = ({ item }: IRootState) => ({
  itemEntity: item.entity
});

const mapDispatchToProps = { getEntity };

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ItemDetail);
