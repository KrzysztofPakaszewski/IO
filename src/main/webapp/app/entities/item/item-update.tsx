import React from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps } from 'react-router-dom';
import { Button, Row, Col, Label } from 'reactstrap';
import { AvFeedback, AvForm, AvGroup, AvInput, AvField } from 'availity-reactstrap-validation';
import { ICrudGetAction, ICrudGetAllAction, setFileData, openFile, byteSize, ICrudPutAction } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IRootState } from 'app/shared/reducers';

import { IUser } from 'app/shared/model/user.model';
import { getUsers } from 'app/modules/administration/user-management/user-management.reducer';
import { getEntity, updateEntity, createEntity, setBlob, reset } from './item.reducer';
import { IItem } from 'app/shared/model/item.model';
import { convertDateTimeFromServer, convertDateTimeToServer } from 'app/shared/util/date-utils';
import { mapIdList } from 'app/shared/util/entity-utils';

export interface IItemUpdateProps extends StateProps, DispatchProps, RouteComponentProps<{ id: string }> {}

export interface IItemUpdateState {
  isNew: boolean;
  ownerId: string;
}

export class ItemUpdate extends React.Component<IItemUpdateProps, IItemUpdateState> {
  constructor(props) {
    super(props);
    this.state = {
      ownerId: '0',
      isNew: !this.props.match.params || !this.props.match.params.id
    };
  }

  componentWillUpdate(nextProps, nextState) {
    if (nextProps.updateSuccess !== this.props.updateSuccess && nextProps.updateSuccess) {
      this.handleClose();
    }
  }

  componentDidMount() {
    if (this.state.isNew) {
      this.props.reset();
    } else {
      this.props.getEntity(this.props.match.params.id);
    }

    this.props.getUsers();
  }

  onBlobChange = (isAnImage, name) => event => {
    setFileData(event, (contentType, data) => this.props.setBlob(name, data, contentType), isAnImage);
  };

  clearBlob = name => () => {
    this.props.setBlob(name, undefined, undefined);
  };

  saveEntity = (event, errors, values) => {
    if (errors.length === 0) {
      const { itemEntity } = this.props;
      const entity = {
        ...itemEntity,
        ...values
      };

      if (this.state.isNew) {
        this.props.createEntity(entity);
      } else {
        this.props.updateEntity(entity);
      }
    }
  };

  handleClose = () => {
    this.props.history.push('/item');
  };

  render() {
    const { itemEntity, users, loading, updating } = this.props;
    const { isNew } = this.state;

    const { image, imageContentType } = itemEntity;

    return (
      <div>
        <Row className="justify-content-center">
          <Col md="8">
            <h2 id="culexApp.item.home.createOrEditLabel"><b><i>{itemEntity.title}</i></b> </h2>
          </Col>
        </Row>
        <Row className="justify-content-center">
          <Col md="8">
            {loading ? (
              <p>Loading...</p>
            ) : (
              <AvForm model={isNew ? {} : itemEntity} onSubmit={this.saveEntity}>
                <AvGroup>
                  <AvGroup>
                    <Label id="imageLabel" for="image">
                    </Label>
                    <br />
                    {image ? (
                      <div>
                        <a onClick={openFile(imageContentType, image)}>
                          <img src={`data:${imageContentType};base64,${image}`} style={{ maxHeight: '100px' }} />
                        </a>
                        <br />
                        <Row>
                          <Col md="11">
                          </Col>
                          <Col md="1">
                            <Button color="danger" onClick={this.clearBlob('image')}>
                              <FontAwesomeIcon icon="times-circle" />
                            </Button>
                          </Col>
                        </Row>
                      </div>
                    ) : null}
                    <input id="file_image" type="file" onChange={this.onBlobChange(true, 'image')} accept="image/*" />
                    <AvInput type="hidden" name="image" value={image} />
                  </AvGroup>
                </AvGroup>
                <AvGroup>
                  <Label id="titleLabel" for="item-title">
                    Title
                  </Label>
                  <AvField id="item-title" type="text" name="title" />
                </AvGroup>
                <AvGroup>
                  <Label id="categoryLabel" for="item-category">
                    Category
                  </Label>
                  <AvInput
                    id="item-category"
                    type="select"
                    className="form-control"
                    name="category"
                    value={(!isNew && itemEntity.category) || 'Movies'}
                  >
                    <option value="Movies">Movies</option>
                    <option value="Games">Games</option>
                    <option value="Books">Books</option>
                  </AvInput>
                </AvGroup>
                <AvGroup>
                  <Label id="stateLabel" for="item-state">
                    State
                  </Label>
                  <AvField id="item-state" type="text" name="state" />
                </AvGroup>
                <AvGroup>
                  <Label id="preferedDeliveryLabel" for="item-preferedDelivery">
                    Prefered Delivery
                  </Label>
                  <AvInput
                    id="item-preferedDelivery"
                    type="select"
                    className="form-control"
                    name="preferedDelivery"
                    value={(!isNew && itemEntity.preferedDelivery) || 'Courier'}
                  >
                    <option value="Courier">Courier</option>
                    <option value="InPost">InPost</option>
                    <option value="PersonalCollection">PersonalCollection</option>
                  </AvInput>
                </AvGroup>
                <AvGroup>
                  <Label id="preferencesLabel" for="item-preferences">
                    Preferences
                  </Label>
                  <AvField id="item-preferences" type="text" name="preferences" />
                </AvGroup>
                <AvGroup>
                  <Label id="hashLabel" for="item-hash">
                    Hash
                  </Label>
                  <AvField id="item-hash" type="text" name="hash" />
                </AvGroup>
                <Button tag={Link} id="cancel-save" to={`/item/`} replace color="info">
                  <FontAwesomeIcon icon="arrow-left" />
                  &nbsp;
                  <span className="d-none d-md-inline">Back</span>
                </Button>
                &nbsp;
                <Button color="primary" id="save-entity" type="submit" disabled={updating}>
                  <FontAwesomeIcon icon="save" />
                  &nbsp; Save
                </Button>
              </AvForm>
            )}
          </Col>
        </Row>
      </div>
    );
  }
}

const mapStateToProps = (storeState: IRootState) => ({
  users: storeState.userManagement.users,
  itemEntity: storeState.item.entity,
  loading: storeState.item.loading,
  updating: storeState.item.updating,
  updateSuccess: storeState.item.updateSuccess
});

const mapDispatchToProps = {
  getUsers,
  getEntity,
  updateEntity,
  setBlob,
  createEntity,
  reset
};

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ItemUpdate);
