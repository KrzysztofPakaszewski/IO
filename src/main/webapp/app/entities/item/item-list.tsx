import React from 'react';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import { ItemCompMinimized } from './item-component-minimized';


export const ItemList = (itemList) =>{

  return (
      <Grid container spacing={2}>
        {itemList.map( (item, index)=>(
          <Grid item key={index}>
            {ItemCompMinimized(item)}
          </Grid>
        ))}
      </Grid>
  );
}
