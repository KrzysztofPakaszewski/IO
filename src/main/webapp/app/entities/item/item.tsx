import React from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps } from 'react-router-dom';
import { Button, Col, Row, Table } from 'reactstrap';
import { openFile, byteSize, ICrudGetAllAction, getSortState, IPaginationBaseState, JhiPagination, JhiItemCount } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import {Checkbox} from "@material-ui/core";
import { IRootState } from 'app/shared/reducers';
import { getEntities, getEntity } from './item.reducer';
import { IItem } from 'app/shared/model/item.model';
import { APP_DATE_FORMAT, APP_LOCAL_DATE_FORMAT } from 'app/config/constants';
import { ITEMS_PER_PAGE } from 'app/shared/util/pagination.constants';
import {getCurrentlyLoggedUser, getUser} from "app/modules/administration/user-management/user-management.reducer";

import { ItemList } from './item-list';

export interface IItemProps extends StateProps, DispatchProps, RouteComponentProps<{ url: string }> {}

export type IItemState = IPaginationBaseState;

export class Item extends React.Component<IItemProps, IItemState> {
  state: IItemState = {
    ...getSortState(this.props.location, ITEMS_PER_PAGE)
  };

  componentDidMount() {
    this.getEntities();
  }

  sort = prop => () => {
    this.setState(
      {
        order: this.state.order === 'asc' ? 'desc' : 'asc',
        sort: prop
      },
      () => this.sortEntities()
    );
  };

  sortEntities() {
    this.getEntities();
    this.props.history.push(`${this.props.location.pathname}?page=${this.state.activePage}&sort=${this.state.sort},${this.state.order}`);
  }

  handlePagination = activePage => this.setState({ activePage }, () => this.sortEntities());

  getEntities = () => {
    const { activePage, itemsPerPage, sort, order } = this.state;
    this.props.getEntities(activePage - 1, itemsPerPage, `${sort},${order}`);
  };

  render() {
    const { itemList, match, totalItems } = this.props;
    return (
      <div>
        <h2 id="item-heading">
          All items
          <Link to={`${match.url}/new`} className="btn btn-primary float-right jh-create-entity" id="jh-create-entity">
            <FontAwesomeIcon icon="plus" />
            &nbsp; Create new item
          </Link>
        </h2>
          {itemList && itemList.length > 0 ? (
            ItemList(itemList)
          ) : (
            <div className="alert alert-warning">No Items found</div>
          )}
        <div className={itemList && itemList.length > 0 ? '' : 'd-none'}>
          <Row className="justify-content-center">
            <JhiItemCount page={this.state.activePage} total={totalItems} itemsPerPage={this.state.itemsPerPage} />
          </Row>
          <Row className="justify-content-center">
            <JhiPagination
              activePage={this.state.activePage}
              onSelect={this.handlePagination}
              maxButtons={5}
              itemsPerPage={this.state.itemsPerPage}
              totalItems={this.props.totalItems}
            />
          </Row>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (storeState: IRootState) => ({
  itemList: storeState.item.entities,
  totalItems: storeState.item.totalItems,
  item: storeState.item.entity,
  loggedUser: storeState.userManagement.user
});

const mapDispatchToProps = {
  getEntities,
  getCurrentlyLoggedUser,
  getEntity
};

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Item);
