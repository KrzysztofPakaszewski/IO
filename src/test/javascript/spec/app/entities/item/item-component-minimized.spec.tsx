import React from 'react';
import { shallow } from 'enzyme';
import { Button} from 'reactstrap';

import {ItemCompMinimized} from 'app/entities/item/item-component-minimized';


describe('ItemComponentMinimized', () => {
  let mountedWrapper;

  const toggle = ()=>{

  };

  const devProps = {
    id: 23,
    title: "title"
  };


  const wrapper = (props = devProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<ItemCompMinimized {...props} />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  // All tests will go here
  it('Renders an Item Comp Minimized without image', () => {
    const component = wrapper();
    // the created snapshot must be committed to source control
    expect(component).toMatchSnapshot();
    expect(component.find(Button).length).toEqual(1);
  });
});
