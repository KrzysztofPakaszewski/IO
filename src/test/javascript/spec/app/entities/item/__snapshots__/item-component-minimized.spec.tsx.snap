// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemComponentMinimized Renders an Item Comp Minimized without image 1`] = `
<WithStyles(ForwardRef(Card))>
  <Button
    color="link"
    tag={[Function]}
    to="/search/23"
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
      justify="space-around"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      />
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        title
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </Button>
</WithStyles(ForwardRef(Card))>
`;
